tools/clang/unittests/HLSLExec/LongVectors.cpp

```diff
@@ -1189,8 +1206,16 @@ void dispatchTest(ID3D12Device *D3DDevice, bool VerboseLogging,
   std::vector<size_t> InputVectorSizes;
   if (OverrideInputSize)
     InputVectorSizes.push_back(OverrideInputSize);
-  else
-    InputVectorSizes = {3, 5, 16, 17, 35, 100, 256, 1024};
+  else {
+    InputVectorSizes = {3, 5, 16, 17, 35, 100, 256};
+    size_t MaxInputSize = 1024;
+    if (IsStructuredBufferLoadAndStoreOp(OP))
+      // StructuredBuffers are capped at 2048 bytes.
+      MaxInputSize = 2048 / sizeof(T);
+
+    if (InputVectorSizes.back() < MaxInputSize)
```

Comment:
I guess this avoids duplicating a test of the maximum input size, but what happens if `InputVectorSizes.back() > MaxInputSize`? Is that really something we want to test?

Reply:
Won't ever hit now. But its a fair point to keep this framework robust to future test cases that might test some larger user defined type.

I'll update it.